presto-verifier/src/main/java/com/facebook/presto/verifier/framework/DataVerification.java

```diff
@@ -161,7 +169,19 @@ else if (QUERY_BANK_MODE.equals(runningMode)) {
     @Override
     protected DeterminismAnalysisDetails analyzeDeterminism(QueryObjectBundle control, DataMatchResult matchResult)
     {
-        return determinismAnalyzer.analyze(control, matchResult.getControlChecksum());
+        if (isUseTestChecksumInDeterminismAnalyzer) {
+            // If we are told to use test checksum for determinism analysis then helper action points to the test,
+            // so we run determinism analysis with test checksum
+            DeterminismAnalysisDetails analysis = determinismAnalyzer.analyze(control, matchResult.getTestChecksum(), getTestAction());
+            if (analysis.getDeterminismAnalysis().isDeterministic()) {
+                return analysis;
+            }
+            // If determinism analysis on test decided that query is non-deterministic, then double check that on the control,
+            // in case test side returns non-deterministic different (from control) results.
+            return determinismAnalyzer.analyze(control, matchResult.getControlChecksum(), getControlAction());
+        }
+        // Default behavior - we run determinism analysis, which uses helper action with control checksum.
+        return determinismAnalyzer.analyze(control, matchResult.getControlChecksum(), null);
```

Comment:
I'm not sure it makes sense to ever use the "helper" action here.  if you want to check if something is deterministic, you need to run it in the original environment. We should change the behavior and always pass the control action.  @singcha will this break anything for presto-on-spark verification?

Reply:
@rschlussel
I understand what you are saying and was quite surprised to see that we use helper action to do determinism analysis.
However, this change is to allow less compute on the control side rather than to fix the determinism analysis.
I would like to keep this change in its scope and avoid changing anything outside of it.
Can we follow this PR up with moving determinism analysis to control by default?